<launch>
    
    <param name="scan_line" type="int" value="32" />

    <!-- if 1, do mapping 10 Hz, if 2, do mapping 5 Hz. Suggest to use 1, it will adjust frequence automaticlly -->
    <param name="mapping_skip_frame" type="int" value="1" />

    <!-- remove too closed points 移除过近的点,这里设置0.3m范围内的激光点被移除-->
    <param name="minimum_range" type="double" value="0.3"/>

    <!-- 水平线的扫描一圈360°,相邻的两点的间距相差0.2°(水平分辨率 = 360/1800), 每条激光线扫描一圈发射1800个点 -->
    <!-- 竖直线的分辨率,跟激光雷达的线数有关,相邻线之间的分辨率为0.4 -->
    <param name="mapping_line_resolution" type="double" value="0.2"/>
    <param name="mapping_plane_resolution" type="double" value="0.4"/>

    <!-- 打开三个节点,点云数据预处理. 激光里程计. 激光建图 -->
    <node pkg="aloam_velodyne" type="ascanRegistration" name="ascanRegistration" output="screen">
        <param name="lidar_topic" type="string" value="/velodyne_points" />
    </node>

    <node pkg="aloam_velodyne" type="alaserOdometry" name="alaserOdometry" output="screen" />

    <node pkg="aloam_velodyne" type="alaserMapping" name="alaserMapping" output="screen" />

    <!-- 打开rviz显示激光雷达建图 -->
    <arg name="rviz" default="true" />
    <group if="$(arg rviz)">
        <node launch-prefix="nice" pkg="rviz" type="rviz" name="rviz" args="-d $(find aloam_velodyne)/rviz_cfg/aloam_velodyne.rviz" />
    </group>

</launch>
